src/osd/OSD.cc

```diff
@@ -2403,6 +2403,21 @@ OSD::OSD(CephContext *cct_,
   trace_endpoint.copy_name(ss.str());
 #endif
 
+  // Determine scheduler type for this OSD
+  auto get_scheduler_type = [&conf = cct->_conf]() {
+    const std::string _type = conf.get_val<std::string>("osd_op_queue");
+    const std::string *type = &_type;
```

Comment:
nit: do we need this pointer at all? How about:

```suggestion
if (auto type = conf.get_val<std::string>("osd_op_queue"); type != "debug_random") {
  return type;
} else {
  // ...
  return index_lookup[which];
}
```

Reply:
Agreed. I have incorporated your suggestion. PTAL.